src/ghazal_azadi_ex/ex02/bubble_sort.py

```diff
@@ -0,0 +1,25 @@
+def bubble_sort(my_data):
+    my_data = list(my_data)
+    # Changing type of data to list which becomes mutable
+    dim = len(my_data)
+    made_swap = True
+    swaps = 0
+    while made_swap:
+        made_swap = False
+        for cnt in range(dim-1):
+            if my_data[cnt] > my_data[cnt+1]:
+                my_data[cnt], my_data[cnt+1] = my_data[cnt+1], my_data[cnt]
+                # Comparing elements and swap them if needed.
+                made_swap = True
+                swaps = swaps + 1
```

Comment:
There's a easier way to solve this problem. The way you do this will make some redundant comparison for sorted data.